src/xwayland/xwm.c

```diff
@@ -205,7 +205,11 @@ read_properties(struct wlc_xwm *xwm, struct wlc_x11_window *win, const xcb_atom_
          // Class && Name
          // STRING == latin1, but we naively just read it as is. For full support we should convert to utf8.
          if (props[i] == XCB_ATOM_WM_CLASS) {
-            wlc_view_set_class_ptr(view, xcb_get_property_value(reply), xcb_get_property_value_length(reply));
+            char *class = xcb_get_property_value(reply);
+            size_t class_total_len = xcb_get_property_value_length(reply);
+
+            size_t class_class_offset = strnlen(class, class_total_len)+1;
+            wlc_view_set_class_ptr(view, class+class_class_offset, class_total_len-class_class_offset);
```

Comment:
>Does that matter, I think wlc_view_set_class_ptr would then just set it to NULL, doesn't it?

Can we guarantee that the value is always 0 terminated? If that assumption is not true then the class + class_class_offset may not result in NULL.

>But yes, we possibly should set it to the first value instead as fallback.

That's probably good idea.